---
title: "Population genetics of *Trichophorum planifolium* structured by inbreeding, not distribution limits"
output:
  bookdown::pdf_document2:
    toc: false
    number_sections: false
bibliography: plantarum.json
csl: american-journal-of-botany.csl
updated: 2022-01-12
author: "Victoria J. Nowell, Song Wang, and Tyler W. Smith"
date: 2021-02-11
---


# Note

This is the source file for our manuscript. It contains all of the writing
and analysis contained in our publication. However, final formatting,
including revisions from co-authors and peer reviewers, were done in a
separate Word MS word document. It's just too big a hassle to copy those
changes back to this document. Consequently, there will be some differences
between this file and the published document. In case of disagreement, the
published manuscript is the canonical report we stand behind.

# Abstract
Peripheral populations are understood to hold special conservation
significance as repositories of divergent genetic diversity. They may also
be at increased risk of extirpation, due to a combination of small size,
isolation, and marginal habitat conditions. We collected genetic,
ecological, and distributional data to evaluate these hypotheses for an
endangered range-edge population of the woodland sedge *Trichophorum
planifolium*. Our data show that, range-wide, *T. planifolium* has
remarkably low within-population diversity (F~ST~ 0.8, 1% heterozygosity).
Most populations are dominated by a small number of distinct multi-locus
genotypes. Population differentiation for this species is much higher than
other outcrossing, wind-pollinated species. However, the extremely low
observed heterozygosity we document is comparable to results from other
woodland sedges. This suggests that the life history of this group of
plants results in high levels of inbreeding. In this context, the somewhat
reduced diversity and limited divergence of the lone remaining Canadian
population is not remarkable, and does not support suggestions that the
population suffers from unusual levels of inbreeding. Despite being located
on the geographic margin of the species' range, the climatic conditions at
the remaining Canadian population are not extreme compared to the rest of
the range in the northeastern US. However, it is geographically isolated,
and the soil conditions at the location are unusual for the species.
Cluster analysis shows that most populations across the range of the
species share relatively homogeneous soil conditions. In constrast, the
Canadian population forms a small outlying cluster with extreme nutrient
values. Thus, while this outlying population does not appear to be
divergent genetically, it's geographic isolation and divergent
environmental conditions indicate it may hold important conservation value.

# Keywords
abundant center hypothesis; 
conservation genetics; 
microsatellites; 
*Trichophorum planifolium*; 
peripheral populations; 
range limits; 
species distribution models

# Introduction

No limit, aim for 8000 words.

```{R init, echo = FALSE, results = "hide", warning = FALSE, message = FALSE, error = FALSE}
library(knitr)
knit("trich-prep.Rmd")
```

<!-- the conservation value of peripheral populations is controversial -->

In high-latitude countries, a high proportion of species identified as
requiring legal protection (i.e., Endangered species) are actually
geographically peripheral populations of species common at lower latitudes
[@YakimowskiEckert_2007; @GibsonEtAl_2009]. For example, in Canada,
approximately 90% of listed Threatened or Endangered plant species are
locally rare representatives of species that reach the northern extent of
their range in Canada, but are abundant further south
[@YakimowskiEckert_2007]. Concern over the investment of conservation
resources into species that are locally rare but globally common has led to
debate over whether, or when, peripheral populations merit national concern
[e.g., @HunterHutchinson_1994; @HampePetit_2005; @GibsonEtAl_2009].

<!-- Peripheral populations may contain important genetic diversity, while
    also facing elevated risk of extirpation -->

Peripheral populations are typically small and geographically isolated due
to increased habitat variability and reduced resource availability at the
range margin [i.e., the abundant center hypothesis, @Brown_1984].
Consequently, they may be at greater risk of extirpation due to genetic
factors (i.e., inbreeding depression) as well as stochastic demographic,
environmental and catastrophic factors [@SouleMills_1998; @BrookEtAl_2002].
These risks may be exacerbated by anthropogenic threats such as habitat
loss and fragmentation [@YoungEtAl_1996]. Nevertheless, peripheral
populations may be important sources of genetic diversity for the species
[@SafrielEtAl_1994; @LeppigWhite_2006; @EckertEtAl_2008], harbouring unique
genetic diversity and/or adaptations to conditions at the range edge
[@LesicaAllendorf_1995; @BunnellEtAl_2004].

<!-- Attempts to confirm the hypothesized relationship between geographic
    distribution and genetic diversity have mixed success -->

However, empirical studies attempting to quantify the genetic
differentiation of peripheral plant populations have produced mixed results
[@EckertEtAl_2008]. The lack of consistent range-edge genetic
differentiation may be attributable in part to species-specific attributes
[@GibsonEtAl_2009]. In plants, life-history characteristics (i.e., breeding
system and dispersal mechanisms) are a major determinant of population
genetic structure, because they strongly impact the rates of gene flow
[@LovelessHamrick_1984; @FrankelEtAl_1995]. Knowledge of the spatial
distribution of intraspecific genetic diversity can guide conservation
efforts for peripheral populations, especially when the reproductive
biology and ecology of the species is not well understood.

@LesicaAllendorf_1995 proposed a framework for evaluating the conservation
value of peripheral populations, identifying population isolation and
divergent environmental conditions as main criteria. In practice, neutral
genetic markers, such as microsatellites, can be used to identify
populations that have experienced restricted gene flow (i.e., are
genetically isolated) [@CrandallEtAl_2000; @Moritz_2002]. However,
microsatellite markers cannot provide information on adaptive divergence of
populations [@HoldereggerEtAl_2006; @TeixeiraHuber_2021]. Therefore, to
capture variation in selection pressures across the species' range,
environmental differences among populations should be assessed
[@AllendorfEtAl_2013].

*Trichophorum planifolium* (Sprengel) Palla, commonly known as Few-flowered
Club-rush or Bashful Bulrush, is a species that is rare at its northern
limit in Canada, but is common further south. It is a perennial woodland
species in the family Cyperaceae, typically occurring on mesic to dry,
often rocky, slopes in association with hardwoods, especially Oaks
(*Quercus* spp.) [@Crins_2002]. The range of *T. planifolium* extends from
Massachusetts west to Ontario and south to Virginia and Kentucky, with
disjunct populations occurring in Missouri and southern Illinois (Figure
\@ref(fig:distmap)).

The only extant locality in Canada is in southern Ontario, in the nature
sanctuaries at Royal Botanical Gardens in Hamilton, Ontario. *Trichophorum
planifolium* was previously known from one other location in Ontario, in
Rouge Valley near Toronto, but no plants have been located at this site
since 2005 [@SmithRothfels_2007]. Due to its limited Canadian range and an
apparent decline in population size, *T. planifolium* is listed as
Endangered under the Canadian Species at Risk Act [SARA, @COSEWIC_2000]. 

*Trichophorum planifolium* is caespitose, growing in dense, low tufts (~ 40
cm high). Individual plants reach a maximum diameter of 20-30 cm, and lack
the long rhizomes present in clonally spreading sedges such as *Carex
pensylvanica* Lamarck [@CrinsRettig_2002]. It is wind-pollinated, with
styles and stamens exerted at anthesis, and has neither a showy perianth or
nectaries [@Crins_1989; @IwanyckiEtAl_2010]. Seeds mature in mid-summer and
disperse in late July to August [@IwanyckiEtAl_2010]. A mechanism for seed
dispersal has not been confirmed [@SmithRothfels_2007], but it has been
suggested that the matting of leaves and stems during the fruiting period
may limit most seed dispersal to within a meter of the parent plants,
leading to the formation of colonies of closely related sibling groups
[@Crins_1989]. It has been suggested that low recruitment observed at the
Hamilton site may be due to limited intrapopulational genetic diversity
[@SmithRothfels_2007].

The main objective of our research was to apply Lesica and Allendorf's
[-@LesicaAllendorf_1995] framework to assess the conservation value of the
last remaining peripheral population of *Trichophorum planifolium* in
Canada. Specifically, we assess genetic diversity (richness and divergence)
using neutral microsatellite markers, to evalute the hypotheses that this
peripheral population has reduced richness and increased divergence
relative to central populations. Additionally, we assess variation in
abiotic conditions (climate niche and local soil properties) and spatial
isolation, to test the hypotheses that peripheral populations are subject
to divergent environmental conditions and reduced connectivity to core
populations.

# MATERIALS AND METHODS
## Population Sampling

In spring 2014, we sampled 29 populations of *T. planifolium* across its
geographic distribution (Figure \@ref(fig:distmap), Table
\@ref(tab:sumtab)). An additional population sample was previously obtained
in 2011 in Kentucky. At each site, we sampled individuals haphazardly, with
at least 1 m between individuals, and a maximum distance of 100 m between
the most distant samples. In total, 919 tissue samples were obtained, with
a minimum of four and a maximum of 50 samples per population (mean = 31.7).
Our protocol was modified for one very small population (PACR), where one
sample was taken at a distance of 150 m to increase sample size to seven. A
single herbarium voucher was collected per population, except for the
locations in Ontario, New Hampshire, and West Virginia, where the species
is Endangered and collection of whole plants is prohibited. Vouchers have
been deposited at DAO [abbreviations follow @Thiers_2020].

Tissue samples were immediately placed on silica gel and stored
at room temperature prior to molecular analysis. A random subsample of 20
samples per population were selected for genetic analysis. For populations
with fewer than 20 samples, all individuals were used.

```{R distmap, fig.width = 7.25, fig.height = 3.5, dev = 'jpeg', fig.show = 'hold', dpi = 200, echo = FALSE, fig.cap = "*Trichophorum planifolium* distribution. A. Populations Sampled: Grey points are GBIF records; red diamonds are populations sampled for the genetic survey and soil analysis. B. Maxent species distribution model. Highest suitability is indicated in green, declining through yellow, orange and grey."}
par(mar = c(0.1,0.1,0.1,0), mfrow = c(1, 2))

plot(st_geometry(na.simp), xlim = c(zm@xmin, zm@xmax),
     ylim = c(zm@ymin, zm@ymax), border = "grey",
     bgc = "lightblue", col = "white")  
plot(gl.proj, add = TRUE, border = "grey", col = "lightblue")
points(trich.proj, pch = 16, col = "grey30", cex = 0.7)
points(samples.proj, cex = 1, pch = 23, bg = 'red')
box()

text("A", x = grconvertX(0.1, from="npc", to="user"), 
     y = grconvertY(0.93, from = "npc", to="user"), cex = 2)

par(mar = c(0.1,0,0.1,0.1))

plot(st_geometry(na.simp), xlim = c(zm@xmin, zm@xmax),
     ylim = c(zm@ymin, zm@ymax), border = "grey",
     bgc = "lightblue", col = "white")  
plot(predTR, add = TRUE, legend = FALSE)
plot(gl.proj, add = TRUE, border = "grey", col = "lightblue")
box()

text("B", x = grconvertX(0.1, from="npc", to="user"), 
     y = grconvertY(0.93, from = "npc", to="user"), cex = 2)

```


(ref:table1cap) *Trichophorum planifolium* populations sampled. Population:
the population code, the first two letters indicate the US state or
Canadian province. Lat and Lon: geographic coordinates. N: total samples
per population (samples with all 11 loci scored in parentheses). Ap:
private alleles. Ar: mean allelic richness per locus. MLG: multilocus
genotypes. eMLG: multilocus genotypes in rarefied samples of 10
individuals. Hs: gene diversity. Ho: observed heterozygosity. F~IS~:
population inbreeding coefficient. F~D~ : mean pairwise F~ST~ for the
population when compared to all other populations. The value in the Total
row for the F~D~ column is the global F~ST~ value.

```{R sumtab, results = 'asis', echo = FALSE}
sumTable <- as.data.frame(samples)
sumTable <- sumTable[, c("population", "Latitude", "Longitude", "n",
                        "NComp", "mlg", "emlg", "meanFst", "Ap", "Ar",
                        "Ho", "Hs", "Fis")]
sumTable["KY", "population"] <- "KYLE"
rownames(sumTable) <- NULL

tmp <- sumTable[1,]
tmp[, names(total)] <- total
tmp[, "population"]  <- "Total"
sumTable <- rbind(sumTable, tmp)
sumTable[sumTable$population == "Total",
         !colnames(sumTable) %in% names(total)] <- NA

options(knitr.kable.NA = '')

sumTable %>%
  mutate(Latitude = round(Latitude, 4),
         Longitude = round(Longitude, 4),
         N = paste(n, "(", NComp, ")", sep = ""),
         emlg = round(emlg, 2),
         Fst = round(meanFst, 2),
         Ar = round(Ar, 2),
         Ho = round(Ho, 2),
         Hs = round(Hs, 2),
         Fis = round(Fis, 2)
         ) %>%
  dplyr::select(population, Latitude, Longitude,
         N, Ap, Ar, mlg, emlg, Hs, Ho, Fis, Fst) %>%
  kable(caption = '(ref:table1cap)')
```

## Genetic Analysis

For each individual, 20 mg of dried leaf tissue was ground with 3
mm-diameter stainless steel beads in a 2 mL Eppendorf tube at 30 Hz for 2
min using a TissueLyser II (Qiagen, Venlo, Limburg, Netherlands). Total
genomic DNA was extracted from ground tissue using the Nucleospin Plant II
Kit (Machery-Nagel, Düren, Germany). The quality and concentration of DNA
obtained was verified using a Nanodrop 2000 UV-Vis Spectrophotometer
(ThermoScientific, Waltham, Massachusetts, U.S.A.).

We developed eleven nuclear microsatellite markers [see @NowellEtAl_2015
for details] to assess the population genetics of *T. planifolium*.
Amplification of microsatellite fragments were performed in 8 µl reaction
volumes containing 0.416 µL primer mix (10 mM; 10:1 untagged to tagged
primer), 0.192 µL dye-labelled CAG Tag (10 mM; 6-FAM or VIC, Life
Technologies, Carlsbad, California, U.S.A.), 0.24 µL DMSO, 4 µL 2X Phusion
High-Fidelity Master Mix with HF Buffer (New England Biolabs, Ipswich,
Massachusetts, U.S.A.), 2.152 µL ddH2O and 1 µL DNA (10 ng/µL) using a
T-100 Thermal Cycler (Bio-Rad, Hercules, California, U.S.A.). Cycling
conditions followed Touchdown-TD PCR [@KorbieMattick_2008] as in
@NowellEtAl_2015. PCR products were visualized on a 1% agarose gel
stained with GelRed (Biotium, Hayward, California, U.S.A.) and viewed with
a High Performance Transilluminator (UVP, Upland, California, U.S.A.) with
a 100 bp DNA ladder (New England Biolabs) to confirm the presence and size
of amplification products and absence of contamination prior to genotyping.

Amplification products were subsequently pooled into groups of four and
visualized by capillary electrophoresis using a 3130xL Genetic Analyzer
(Life Technologies) with the GeneScan 500 LIZ Size Standard (Life
Technologies). Individual samples were genotyped using Genemapper v.5
software (Life Technologies) and verified with manual scoring.

Due to amplification failures, we were unable to score all samples for all
11 markers. We retained only those samples for which we could score at
least 8 loci. We enumerated private alleles, calculated allelic richness
[@ElMousadikPetit_1996; @AdamackGruber_2014], gene diversity, observed
heterozygosity and the fixation index F~IS~ [@Nei_1987;
@GoudetJombart_2015] for each population. Tests for departures from
Hardy-Weinberg equilibrium (HWE) were performed with the R package 'pegas'
ver. 0.13 [@Paradis_2010].

We counted the number of distinct multi-locus genotypes (MLG) for each
population, using the methods provided by @KamvarEtAl_2015. When samples
with incomplete genotypes could not be unambiguously assigned to a single
MLG, we randomly assigned them to a matching MLG in same population, if
present. To account for differences in sample size, we also calculated
rarefied MLG counts for each population, as a standardized measure of
genetic richness. The rarefaction estimates were calculated for subsamples
of 10 individuals per population [referred to as eMLG hereafter,
@OksanenEtAl_2019]. Populations with fewer than 10 invidivduals scored were
excluded from rarefaction, and all analyses based on eMLG.

We also assessed genetic divergence, using an index based on population
pairwise-F~ST~ values. We calculated our divergence index (F~D~) as the
mean of pairwise-F~ST~ values for a given population, when compared with
all other populations.

Preliminary review of the data revealed excessive levels of homozygosity,
and no locus was in Hardy-Weinburg equilibrium. Five populations had a
single MLG shared by all individuals and no heterozygosity (Table
\@ref(tab:sumtab)). As discussed below, this indicates high levels of
selfing and/or inbreeding among close relatives. As such, analyses based on
assumptions of Hardy-Weinburg equilibrium, linkage equilibrium and
panmictic populations, such as STRUCTURE [@PritchardEtAl_2000], are
inappropriate for our data. Accordingly, we used multivariate ordinations
to visualize genetic structure among populations [@JombartEtAl_2008].
Similarly, we did not test for the presence of null alleles, as we are not
aware of a test that can distinguish between excessive homozygosity due to
the presence of null alleles, and excessive homozygosity to due to high
levels of inbreeding [e.g., @Brookfield_1996].

We assessed spatial structure using spatial Principal Components Analysis
[i.e., sPCA: @JombartEtAl_2008; @ThioulouseEtAl_2018]. We defined the
spatial neighbourhood of a sample as all other samples within 650 km, with
the strength of each link weighted inversely by distance (via function
`spdep::dnearneigh` with the upper distance bound set to 650). We chose 650
km to ensure there were no disconnected subgroups (i.e., there was at least
one link between the Missouri populations and the eastern populations). We
tested for statistically significant spatial autocorrelation using a
randomization test (function `global.rtest` in R package `adegenet`).

## Soil Sampling and Analysis

Soil samples, composed of at least 7 sub-samples, were collected at each
population (except WVMO and VASC, where the substrate was too rocky for
soil collection, and KYLE, for which tissue samples were collected several
years prior to the development of sampling design implemented here), with
sub-samples made to rooting depth (~10 cm) immediately adjacent to *T.
planifolium* plants. Composite samples were submitted to the University of
Massachusetts Soil Laboratory (American Samples) or the University of
Guelph Soil Laboratory (the Canadian sample), and soil properties including
pH, cation exchange capacity (CEC), organic matter (OM) and soil nutrients
(Calcium, Magnesium, Manganese, Phosphorous, Potassium, Zinc) were
quantified.

We used hierarchical clustering to visualize differences in soil
conditions. Each variable was scaled to mean = 0, standard deviation = 1,
and the Euclidean distance among sites was used to construct UPGMA cluster
dendrograms [@LegendreLegendre_2012]. We validated the dendrogram structure
using the cophenetic correlation coefficient, and identified the optimal
number of clusters using the Kelley-Gardner-Sutcliffe penalty function
[@KelleyEtAl_1996].

## Geographic and Ecological Analysis

We used herbarium records from @GBIF.org_2019 to characterize the
geographic isolation and climatic distribution of *T. planifolium*. To
control for collection bias [see @RadosavljevicAnderson_2014], we used the
`spThin` package [@Aiello-LammensEtAl_2015] to create a thinned data set,
in which retained samples were at least 10 km from all other samples.

We quantified the isolation of each occurence in the thinned GBIF data as
the fitted value of a kernel density smoother fit to those data, the
"Utilization Distribution" of @Worton_1989, and implemented in the R
package `adehabitat` [@Calenge_2006]. This approach incorporates both
distance and density of occurences near a point, with low values (near
zero) assigned to isolated points, and increasing values for points with
more & closer neighbours. There is no upper bound to this value, but
records used in our study had values from 0.007 (most isolated) to 0.113
(least isolated; see results).

To quantify climate suitability for observations, we constructed a Species
Distribution Model (SDM) using a Maxent [@PhillipsEtAl_2017]. We limited
the study extent to 31N-50N degrees latitude, 66W-98W degrees longitude;
this area was chosen to include a 500 km buffer around a minimum convex
polygon containing all GBIF records for *T. planifolium*. Background
training data was limited to level 1 Ecoregions [@OmernikGriffith_2014] in
which *T. planifolium* is present within this extent. The following
environmental layers were considered: 19 bioclimatic variables [WorldClim:
@FickHijmans_2017, www.worldclim.org], aridity [@TrabuccoZomer_2019], and
terrain ruggedness. Terrain ruggedness was calculated from SRTM DEM data
[@JarvisEtAl_2008] with 3-arc second (approximately 90m) horizontal
resolution. We used the Terrain Ruggedness Index in QGIS [@QGIS.org_2020].
The resulting raster was upscaled to match the 30 arc-second (~1km)
resolution of the other layers, using the "average" resampling method. To
reduce collinearity among variables, We selected a subset of these layers
such that the variance inflation factor was less than 10 [function `vif` in
the R package `usdm`: @NaimiEtAl_2014]. The final set of variables included
aridity, WorldClim Bio2 (mean diurnal temperature range), Bio6 (minimum
temperature of coldest month), Bio13 (precipitation of wettest month), and
Terrain Ruggedness. Maxent model parameters were tuned with the
`ENMevaluate` function from the package `ENMeval` [@MuscarellaEtAl_2014].
We used the fitted cumulative log-log values from the SDM as an index of
climate suitability for each record.

All analyses were performed in R version 4.0 [@RCoreTeam_2020], unless
otherwise specified.

# RESULTS
## Genetics
We scored a total of 505 individual genotypes (mean 16.8 per population)
for at least 8 loci, and completed all 11 loci for 408 individuals (Table
\@ref(tab:sumtab)). 46 alleles were detected (mean 4.2 per locus, range
2--9), and a total of 115 unique multi-locus genotypes (MLG) were
described. All but three MLG were confined to a single population. One MLG
was shared by two Missouri populations; one was shared by two Ohio
populations; and one MLG was found in three populations, one each in
Connecticut, Maryland, and eastern New York.

465 (92.1%) individuals were homozygous for all alleles, with only 40
individuals heterozygous at one or more loci. Of the 25 populations with at
least 10 individuals genotyped, the mean eMLG (i.e., genotypes per 10
individuals) was 3.1 (range 1--6.5, Figure \@ref(fig:sumplots)). The
Canadian population had an eMLG of 1.37, corresponding to the 20th
percentile of the distribution. 

The pair-wise F~ST~ values ranged from 0 to 1 (mean 0.79). A group of five
populations shared mutual pairwise-F~ST~ values of exactly 1: CTTP, MAST,
OHBH, OHSC, VAHB. This is a consequence of each of these populations being
fixed for a different homozygous MLG. Ohio populations OHBH and OHPR had a
pairwise F~ST~ of -0.01 [corrected for heterozygosity,
@GoudetJombart_2015], as a consequence of the single MLG present in OHBH
(all 13 samples), also occuring in 16 of 18 samples from OHPR.

The mean genetic divergence, F~D~, was 0.79 (range 0.59--0.9, Figure
\@ref(fig:sumplots)). F~D~ for the Ontario population was 0.87,
corresponding to the 80th percentile of the distribution. The least
divergent populations (in increasing order) were MOBU, WVFM, NYTS, NHER and
PAGP (F~D~ = 0.60 -- 0.68). The most divergent populations (in decreasing
order) were NYMP, VAHB, MAST, CTTP, OHSC, ONRB (F~ST~ 0.87--0.9).

```{R sumplots, fig.width = 6, fig.height = 3, dev = 'jpeg', dpi = 200, echo = FALSE, fig.show = "hold", fig.cap = "Distribution of rarified multi-locus genotypes (eMLG) and population mean pair-wise Fst values (F_D). The red line indicates the population in Ontario, Canada."}
#dev.new(height = 3.5, width = 7)

par(mfrow = c(1, 2))
par(mar = c(3.5, 3.5, 0.5, 0.5))
hist(samples$emlg, breaks = 12, main = "", xlab = "", ylab = "",
     axes = FALSE) 
abline(v = data.frame(samples)["ONRB", "emlg"], col = 2, lwd = 4)

axis(side = 2, cex.axis = 0.7, tcl = -0.33, padj = 0.75)
mtext(side = 2, line = 2, "Frequency")
axis(side = 1, cex.axis = 0.7, tcl = -0.33, padj = -0.75)
mtext(side = 1, line = 2, "eMLG")

par(mar = c(3.5, 1.5, 0.5, 0.5))
hist(samples$meanFst, breaks = 12, main = "", xlab = "",
     ylab = "", axes = FALSE)
abline(v = data.frame(samples)["ONRB", "meanFst"], col = 2, lwd = 4)

axis(side = 2, cex.axis = 0.7, tcl = -0.33, padj = 0.75)
axis(side = 1, cex.axis = 0.7, tcl = -0.33, padj = -0.75)
mtext(side = 1, line = 2, text = expression('F'[D]))

```

There was significant spatial structure in the genetic data, as determined
by a Monte Carlo test of Morin's I (function `adespatial::global.rtest`, p
< 0.001). This is largely due to the relative divergence of the Missouri
populations (Figure \@ref(fig:PCoA)). On the PCoA plot, the Missouri
samples form a discrete grouping on the left side; the single Maryland
population is somewhat separated in the upper right corner, and the Ontario
samples cluster with the Missouri grouping on the lower left. The other
populations are intermixed in the center of the plot. A similar pattern is
shown in the sPCA map, which shows the Missouri samples have low values on
sPCA axis 1, compared to higher values for the eastern populations. The
populations in Ohio, in the center of the geographic range show a mix of
'eastern' and 'western' affinities.

```{R PCoA, fig.width = 7, fig.height = 4, dev = 'jpeg', dpi = 200, echo = FALSE, fig.show = "hold", fig.cap = "Spatial Structure of Genetic Diversity in *Trichophorum planifolium*. Left: Principal Coordinates Analysis of 97 unique microsatellite MLGs. The centroid of all samples from each state or province are labelled, connected to each of the unique MLGs for that jurisdiction. The strongest geographic structure is the clustering of Missouri samples on the left (MO, green squares), Maryland on the right (MD, red circles) and Ontario at the lower left (ON, blue diamonds). Right. sPCA map. The color for each population indicates its position along the first axis of the sPCA ordination."}
#dev.new(height = 4, width = 7)

par(mfrow = c(1, 2))
par(mar = c(3, 3, 0.5, 0.1))
plot(scores(clonePCoA)[, c(1,2)], type = 'n',
     asp = 1, axes = FALSE, xlab = NA, ylab = NA)
statesSpider <- unique(unclass(ordispider(clonePCoA, groups = myStates,
                                         col = "grey")))
points(scores(clonePCoA)[, c(1,2)], pch = c(21:24)[myPops],
     asp = 1, bg = c(2:4, "grey")[myPops],
     col = c(1, 1, 1, 0)[myPops],
     cex = c(1.5, 1.5, 1.5, 1)[myPops])
text(statesSpider, cex = 1,
     labels = substr(row.names(statesSpider), 0, 2))
axis(1, cex.axis = 0.75, tcl = -0.2, line = 0, mgp = c(3, 0.3, 0),
     col.axis = "grey30", at = c(-1, 0, 1))
axis(2, cex.axis = 0.75, tcl = -0.2, line = 0, mgp = c(3, 0.3, 0),
     las = 1, col.axis = "grey30", at = c(-1, 0, 1))
mtext(side = 1, line = 1.25, cex = 0.9,
      text = paste("PCA Axis 1, ", eigVals[1], "% of Total Variance",
            sep = ""))
mtext(side = 2, line = 1.25,, cex = 0.9,
      text = paste("PCA Axis 2, ", eigVals[2], "% of Total Variance",
            sep = ""))
box()

par(mar = c(3, 2, 0.5, 0.5))

axcol <- trich.spca1$li[, 1] - min(trich.spca1$li[, 1])
axcol <- viridis(100)[round(axcol/max(axcol) * 100)]

plot(st_geometry(na.simp), xlim = c(zm@xmin, zm@xmax),
     ylim = c(zm@ymin, zm@ymax),
     border = "grey", axes = TRUE,
     graticule = st_crs(us),
     bgc = "lightblue", col = "white",
     lat = c(30, 45), lon = c(-75, -90),
     mgp = c(3, 0.5, 0), 
     tcl = -0.2, cex.axis = 0.75, col.axis = "grey30")  #"gray95"
plot(gl.proj, add = TRUE, border = "grey", col = "lightblue")
points(samples.proj, pch = 22, bg = axcol, col = "grey",
       cex = 3)
rect(xleft = grconvertX(seq(0.70, 0.95, length.out = 100),
                        from="npc", to="user"), 
     xright = grconvertX(seq(0.7025, 0.9525, 
                             length.out = 100),
                         from="npc", to="user"),
     ytop = grconvertY(0.1, from="npc", to="user"),
     ybottom = grconvertY(0.05, from="npc", to="user"),
     col = viridis(100), border = NA)
text(y = grconvertY(0.125, from = "npc", to = "user"),
     x = grconvertX(0.97, from = "npc", to = "user"),
     "sPCA Axis 1", pos = 2, cex = 0.75)

```

## Soils

The cophenetic correlation of the UPGMA clustering of soil data was 0.94,
indicating strong correspondance between the dendrogram structure and the
original distance values. The optimal number of clusters identified by the
Kelley-Gardner-Sutcliffe test was four (Figure \@ref(fig:soil-plots)). 22
of 27 assessed populations belonged to the main cluster, with three
outlying clusters containing one or two populations each. The Maryland
population MDDM formed a cluster on its own, characterized by low pH, high
organic matter, and high values of P, K, and Mn. Populations MABM and PAGP
formed a cluster, characterized by high Ca and Mg. Populations ONRB and
PACR formed the fourth cluster, characterized by high Ca and Zn and low Mn
(Table \@ref(tab:soil-table)).

```{R soil-plots, fig.height = 4, fig.width = 7, fig.cap = "Soil Analysis of *T. planifolium* populations. Left: UPGMA dendrogram, based on 9 soil variables. The four clusters identified by the Kelley-Gardner-Sutcliffe test are indicated. Right: Geographic distribution of soil clusters. The plot symbols on the map indicate the soil cluster from the dendrogram; empty circles are populations without soil data.", echo = FALSE}
#dev.new(height = 4, width = 7)
par(mfrow = c(1, 2))
par(mar = c(3, 2, 1, 1))
soilDend <- as.dendrogram(soilClust)
labels_cex(soilDend) <- 0.65
soilDend <- color_labels(soilDend, k = 4)
plot(soilDend, axes = FALSE)
axis(side = 2, cex.axis = 0.65, at = c(0, 4, 8), las = 1,
     hadj = -1, tcl = -0.25, )
mtext(side = 2, "Distance", cex = 0.75, line = 1)
rect.dendrogram(soilDend, k = 4, border = 8,
                lty = 5, lwd = 1)

par(xpd = TRUE)
points(x = 14, y = -1.75, pch = 21, col = 'darkgrey',
       bg = "#767F00", cex = 1.5)
points(x = 24.5, y = -1.75, pch = 22, col = "darkgrey",
       bg = "#009681", cex = 1.5) 
points(x = 1, y = -1.75, pch = 23, col = "darkgrey",
       bg = "#CC476B", cex = 1.5)
points(x = 26.5, y = -1.75, pch = 24, col = "darkgrey",
       bg = "#7866D8", cex = 1.5)
## NB: colors come from rainbow_hcl(n = 4, c = 90, l = 50),
## built-in to color_labels

par(xpd = FALSE)
par(mar = c(0, 0, 0, 0))
plot(st_geometry(na.simp), xlim = c(zm@xmin, zm@xmax),
     ylim = c(zm@ymin, zm@ymax),
     border = "grey", axes = TRUE,
     graticule = st_crs(us),
     bgc = "lightblue", col = "white",
     lat = c(30, 45), lon = c(-75, -90),
     mgp = c(3, 0.5, 0), 
     tcl = 0, cex.axis = 0.75, col.axis = "white")  #"gray95"
plot(gl.proj, add = TRUE, border = "grey", col = "lightblue")

## Adding points separately to control which points are on top
points(samples.proj[c("KY", "WVMO", "VASC"), ], pch = 1, col = 'black', cex = 1.5)
points(samples.proj[names(soilCut)[soilCut == 1], ],
       pch = 21, col = 'darkgrey', bg = "#767F00", cex = 1.5)
points(samples.proj[names(soilCut)[soilCut == 2], ],
       pch = 22, col = "darkgrey", bg = "#009681", cex = 1.75)
points(samples.proj[names(soilCut)[soilCut == 4], ],
       pch = 24, col = "darkgrey", bg = "#7866D8", cex = 1.5)
points(samples.proj[names(soilCut)[soilCut == 3], ],
       pch = 23, col = "darkgrey", bg = "#CC476B", cex = 1.5)

```

```{R soil-table, echo = FALSE, eval = TRUE}
soilData %>% group_by(group) %>%
  summarize(across(.fns = list(mean = mean, sd = sd))) %>%
  t %>% round(1) ->soilTable
soilTable <- soilTable[-1, c(1, 2, 4, 3)]
tableVals <- matrix(character(), ncol = 4)
colnames(tableVals) <- c("Main", "MABM & PAGP", "ONRB & PACR", "MDDM")
for(var in soilCols){
  MEANS <-
    soilTable[paste(var, "_mean", sep = ""), ] %>%
    sprintf("%.1f", .)
  SDS <-
    soilTable[paste(var, "_sd", sep = ""), ] %>%
    sprintf("%.1f", .)
  rowNs <- rownames(tableVals)
  tableVals <- rbind(tableVals, paste(MEANS, SDS,
                                     sep = " ± "))
  rownames(tableVals) <- c(rowNs, var)
}
tableVals[, "MDDM"] <- gsub(" ± NA", "", tableVals[, "MDDM"])

tableVals <- tableVals[c("CEC", "OM", "pH", "Ca", "K", "Mg",
                        "Mn", "P", "Zn"), ]
rownames(tableVals) <- c("CEC (meq/100g)", "OM (%)", "pH",
                        "Ca (ppm)", "K (ppm)", "Mg (ppm)",
                        "Mn (ppm)", "P (ppm)", "Zn (ppm)")

kable(tableVals, caption = '(ref:soilTableCap)')

```

(ref:soilTableCap) Soil analysis for *T. planifolium* populations. Values reported as mean ± standard deviation for each cluster. Note: the MDDM cluster had only one population, so there is no standard deviation to report.

## Geographic Isolation and Climate Suitability 

Population isolation for the herbarium records ranged from 0.00716 to 0.113
(lower values indicate greater isolation) (Figure \@ref(fig:isolation)).
The Canadian population had an isolation index of 0.00718, putting it in
the 1st percentile of the 266 records. Climate suitability for each of the
herbarium records ranged from 1.9e-5 to 4.5e-3; suitability at the Ontario
population scored 3.6e-4, the 30th percentile of the total distribution
(Figures \@ref(fig:distmap) and \@ref(fig:isolation)).

(ref:isolationLabel) Geographic Isolation and Climate Suitability of *T.
planifolium* records. Population Isolation based on the density of
occurences around each GBIF record using the Utilization Distribution;
Climate Suitability presents the fitted values for each record from a
Maxent distribution model. Red lines indicate the values for the last
remaining Canadian population.

```{R isolation, echo = FALSE, fig.cap = '(ref:isolationLabel)', fig.height = 3, fig.width = 6}
#dev.new(height = 3, width = 6)
jpeg(file = "isolation.jpg", width = 6, height = 3,
     units = "in", res = 600)
par(mfrow = c(1, 2))
par(mar = c(3.5, 3.5, 0.5, 0.5))
hist(gbifthin$kernel, xlab = "",
     main = "", ylab = "", axes = FALSE)
abline(v = data.frame(samples["ONRB", ])["kernel"], col = 2,
       lwd = 2)

axis(side = 2, cex.axis = 0.7, tcl = -0.33, padj = 0.75)
mtext(side = 2, line = 2, "Frequency")
axis(side = 1, cex.axis = 0.7, tcl = -0.33, padj = -0.75)
mtext(side = 1, line = 2, "Population Isolation")

par(mar = c(3.5, 1.5, 0.5, 0.5))

hist(gbifthin$suitability, xlab = "", ylab = "", main = "",
     axes = FALSE)
abline(v = data.frame(samples["ONRB", ])["LQH_2"], col = 2,
       lwd = 2)

axis(side = 2, cex.axis = 0.7, tcl = -0.33, padj = 0.75)
axis(side = 1, cex.axis = 0.7, tcl = -0.33, padj = -0.75)
mtext(side = 1, line = 2, text = "Climate Suitability")

dev.off()

```



```{R regression, echo = FALSE, eval = FALSE}
lmp <- function (modelobject) {
  if (class(modelobject) != "lm") stop("Not an object of class 'lm' ")
  f <- summary(modelobject)$fstatistic
  p <- pf(f[1],f[2],f[3],lower.tail=F)
  attributes(p) <- NULL
  return(p)
}

lms <- data.frame(response = rep(c("emlg", "meanFst"), each = 3),
                 explanatory = rep(c("isolation", "marginality",
                                     "suitability"), 2),
                 F = NA, R = NA, p = NA)

for(i in seq_along(lms[, 1])){
  form <- formula(paste(lms$response[i],
                       "~ log(", lms$explanatory[i], ")"))
  mod <- lm(formula = form,
           data = samples@data)
  lms[i, "R"] <- round(summary(mod)$adj.r.squared, 2)
  lms[i, "p"] <- round(lmp(mod), 2)
  lms[i, "F"] <- round(summary(mod)$fstatistic[1], 2)
}

# make sure our distance matrices are aligned:
stopifnot(row.names(as.matrix(trichFst)) == row.names(as.matrix(geogD)))

ibd <- mantel(trichFst, as.dist(geogD))
lms <- lms[c(1, 1:6), ]
lms[1, 1]  <- "IBD"
lms[1, 2]  <- ""
lms[1, "R"] <- round(ibd$statistic, 2)
lms[1, "p"] <- round(ibd$signif, 2)
lms[1, "F"] <- NA

colnames(lms)[1:2] <- c("Dependent", "Independent")
lms[c(3, 4, 6, 7), 1] <- ""
lms[2, 1] <- "eMLG ~"
lms[5, 1] <- "Fst ~"
rownames(lms) <- NULL
## ibdMat <- data.frame(Fst = as.numeric(trichFst),
##                     distance = as.numeric(as.dist(geogD)))
## ggplot(ibdMat) + aes(x = log(Fst), y = distance) +
##   geom_point() + geom_smooth(method = "lm")

lms <- lms[-1, ]
kable(lms,
      caption =
      "Biogeographic Tests. Each line reports linear regression parameters F-statistic, adjusted R-squared value and P-value. Dependent variables: eMLG, rarefied richness of multi-locus genotypes; Fst, mean pairwise Fst values for each population. Independent variables: indexes of isolation, peripherality (see text), and habitat suitability from SDM.")
```

# DISCUSSION
<!-- We found no evidence to support the hypothesis that genetic diversity -->
<!-- declines with geographic peripherality, ecological marginality, or spatial -->
<!-- isolation. However, the highly structured populations (i.e., global F_ST = -->
<!-- 0.8) and significant east-west spatial structure suggest that this lack of -->
<!-- pattern is not due to 'swamping' gene flow from the core of the -->
<!-- distribution homogenizing genetic diversity in peripheral populations. If -->
<!-- that were the case, we would expect to see lower among-population variation -->
<!-- (F_ST), and relatively higher within-population variation (F_IS). -->

<!-- This contrasts with other recent studies that have found significant -->
<!-- relationships between genetic diversity and spatial or environmental -->
<!-- gradients [@SextonEtAl_2016; @Lee-YawEtAl_2018]. Differences in life -->
<!-- history and reproductive biology may explain the difference. -->
<!-- @Lee-YawEtAl_2018 found that genetic diversity declined towards the -->
<!-- geographic range limits of *Arabidopsis lyrata* ssp. *lyrata*, and also -->
<!-- with distance from the species' environmental niche. In addition, their -->
<!-- data shows that environmental conditions declined towards the range -->
<!-- periphery. This supports the abundant center hypothesis [@Brown_1984].  -->

<!-- On the other hand, @SextonEtAl_2016 found no relationship between genetic -->
<!-- diversity and spatial structure in *Mimulus laciniatus*. Instead, their -->
<!-- work documented Isolation by Environment [@SextonEtAl_2014], where genetic -->
<!-- diversity is structured along environmental gradients, presumably a -->
<!-- consequence of natural selection or non-random mating. -->

<!-- Neither pattern appears in our data.  -->

The most striking observation in our study is the extremely low variation
present within populations (mean F~IS~ 0.91, 1% heterozygosity), and high
level of differentiation among populations (F~ST~ = 0.80). *Trichophorum
planifolium* is apparently a wind-pollinated, outcrossing species, but
these values are higher even than typical for selfing species [i.e., the
mean F~ST~ for selfing species with similar life history traits is
0.4--0.57, following @HamrickGodt_1996; @Nybom_2004].

This supports Crins' [-@Crins_1989] hypothesis that the low stature of *T.
planifolium*, and the complex terrain it occupies, limits outcrossing.
Crins also noted that *T. planifolium* is protandrous, presumably an
adaptation to lessen the frequency of self-pollination. However,
@FriedmanBarrett_2009 showed that the spatial and temporal separation of
male and female flowers in woodland sedges (monecious *Carex*) are largely
ineffective at preventing geitonogamy. They studied seven *Carex* species,
many of which co-occur with *T. planifolium* in woodland habitats (pers.
obs) and have similar caespitose habit. They found only one species had an
outcrossing rate greater than 0.1 (three species had no polymorphic
allozymes, which may reflect a long history of inbreeding).

Lacking any obvious adaptations for long-distance seed dispersal, gene flow
via seed dispersal is likely similarly limited in *T. planifolium*. It is
well documented that plant seeds often travel only one or a few metres
[@CainEtAl_2000]. Seed dispersal in *T. planifolium* occurs in late July
and August, during which time the plants' leaves become matted. Crins
[-@Crins_1989] suggested that this would facilitate deposition of the seeds
immediately adjacent to parent plants (i.e., seeds are gravity dispersed),
resulting in small spatial clumps of closely related tussocks. Like
self-pollination, mating among closely related individuals limits
recombination and leads to increased homogeneity of genotypes in the
population [@LovelessHamrick_1984].

In combination, these life history characteristics may explain the low
population genetic diversity documented here as the outcome of founder
effects, followed by population expansion dominated by selfing. This is
consistent with a general pattern noted in other sedges (*Carex*), which
can be divided into two life history categories [@FordEtAl_1998]. "Group 1"
are caespitose, with multiple hermaphroditic culms on each plant, with
flowers in close proximity. This facilitates geitonogamy, resulting in low
intra-population diversity, but high inter-population diversity [e.g.,
@Waterway_1990; @SchellWaterway_1992; @Whitkus_1992, @KullOja_2007].

In contrast, "Group 2" *Carex* are rhizomatous, with more space between
culms from the same individual, and more physical mixing among individuals.
This leads to greater intra-population diversity, and less differentiation
among populations [@FordEtAl_1998].

A caespitose species incapable of rhizomatous spread, *Trichphorum
planifolium* fits the description of *Carex* Group 1, aside from having
bisexual flowers. This likely further exacerbates inbreeding, as it may
engage in both autogamous and geitenogamous selfing. The low levels of
heterozygosity we observed (Ho 0.00--0.03, Table \@ref(tab:sumtab)) are
comparable to values reported for other sedges: *Trichophorum caespitosum*:
Ho = 0.01 to 0.04 [@GodtEtAl_1996], *Carex loliaceae*: Ho = 0.00 to 0.12
[@KullOja_2007] and *Carex magellanica* subsp. *irrigua*: Ho = 0.00 to 0.03
[@KullOja_2010].

The main geographic structure in the genetic data reveals a broad division
between Missouri and the eastern populations (Figure \@ref(fig:PCoA)).
This suggests current *T. planifolium* populations may have emerged from
glacial refugia both in the Appalachian Mountains and on the Ozark plateau
[@DelcourtDelcourt_1991; @Meyer_1997; @SoltisEtAl_2006]. However, if this
is the case, it is interesting to note that populations representing both
eastern and western genetic groups appear in close proximity in Ohio. The
ordination analysis also indicates the Ontario population is genetically
more similar to the Missouri populations than the geographically closer
eastern group.

Gene flow via pollen or seed between Missouri in the west, and Ohio and
Ontario to the east, is unlikely, given the long distances between these
locations (> 500 km). However, the SDM indicates pockets of suitable
environment exist along the Ohio river in southern Indiana and northern
Kentucky (Figure \@ref(fig:distmap)). This may have served as a
post-glacial colonization route linking populations from the two regions;
there may even be overlooked populations of *T. planifolium* in this area,
as this species is easily missed.

The lone remaining Canadian population shows somewhat reduced diversity
(20th percentile for eMLG; Figure \@ref(fig:sumplots)). We have argued
above that population dynamics in this species are characterized by
bottlenecks followed by inbreeding. This is often associated with a
reduction in fitness [@AngeloniEtAl_2011]. However, this cost may be
reduced in self-pollinating species [@Busch_2005;
@HoltsfordEllstrand_1990], and can be further mitigated if the inbreeding
leads to purging of deleterious alleles [@CrnokrakBarrett_2002].

We cannot address this issue directly with our data. Microsatellites are
neutral markers, they do not provide direct evidence of adaptive diversity;
they only provide insight into gene flow and reproductive strategies. We
can say that *T. planifolium* populations with low diversity are relatively
common and widespread (e.g., CT, MA, OH, VA), and several of
them are quite large and apparently stable (personal observation).
Furthermore, this pattern of limited intra-population genetic diversity is
common in other woodland sedges. From this, we conclude that the reduced
genetic diversity of the last Canadian population does not suggest inbreeding
depression likely to be a threat to its persistence.

The Canadian population is moderately divergent genetically (80th
percentile for F~D~; Figure \@ref(fig:sumplots)), and climatic conditions
at that location are not extreme for the species (30th percentile for
suitability). However, it is extremely isolated, and the soil conditions at
the site are shared with only one other population. This provides indirect
evidence that this population may be adapted to site-specific environmental
conditions, indicating elevated conservation value [as per
@LesicaAllendorf_1995]. Indeed, protection of many populations of *T.
planifolium* would be necessary to conserve species-level genetic diversity
as it is chiefly partitioned among, rather than within, populations [see
@BaskaufEtAl_2014 or @KoellingEtAl_2011, for example]. These results also
suggest that risks of outbreeding depression are high with outcrossing, and
future restorative efforts (i.e., augmentation) for the Canadian population
should be avoided or approached with caution [@FrankhamEtAl_2011]. It
should also be considered that mixing of populations with differing genetic
backgrounds could decrease the conservation value of the Canadian
population [@BarrettKohn_1991].

# Acknowledgements
The authors thank the staff of Royal Botanical Gardens in Hamilton,
Ontario, especially Natalie Iwanycki, who provided detailed reports on *T.
planifolium* in Canada. Kasia Dadej (Agriculture and Agri-Food Canada)
assisted with scoring the microsatellite data. The following people
assisted with obtaining access to field sites: George Yaskievych, Rick
Gardner, Tom Wieboldt, P. J. Harmon, Jim Vanderhorst, John Kunsman, Jeff
Saarela, Steve Grund, Steve Glenn, Sara Cairns, Bryan Connolly, Robert
Bertin, Karen Searcy, Rich Ring, Steven Daniel, Greg Iffrig, Brian
Davidson, April Daniels, Dale Egbert, Lynn Prater, Nathan Jester, Rich
McCarty, Dave Minney, Bill Stanely, Phil Hoffman, Glen Cobb, Levi Miller,
Pearl's family, Bruce Metz, Fawn Kearns, Barbara Sargent, Tom Metzger, the
Morris County Parks Commission, Becky Petro, Julie Richburg, Nancy Putnam,
Jesse Jaycox, and Mark Quinn. Funding for this project was provided by the
Ontario Ministry of Natural Resources Species at Risk Stewardship Fund
(SAR-SF), the Environment and Climate Change Canada Interdepartmental
Recovery Fund (IRF), and the Carleton University Bonneville Family
Scholarship.

# AUTHOR CONTRIBUTIONS

VJN and TWS conceived the study. VJN designed the field sampling, and VJN
and TWS completed the sample collection. VJN and SW completed the lab
experiments. VJN completed the analysis, with direction and supervision
from TWS. VJN completed the first draft of the manuscript. TWS provided
additional analyses and completed the final draft of the manuscript.

# Data Availability

This manuscript and all associated data is available at the GitHub
repository, https://github.com/plantarum/trich for inspection during peer
review. This repository will be transfered to Zenodo for permanent, public
archiving after completion of the peer review process.


# Extra Figure

Added during revision

```{R popsize, fig.width = 6, fig.height = 3, dev = 'jpeg', dpi = 200, echo = FALSE, fig.show = "hold", fig.cap = "Genetic diversity (eMLG) and divergence by population size. Very small: < 100 individuals; Small: 100-500; Medium: 500-1000; Large: 1000+."}
#dev.new(height = 3, width = 7)
## jpeg(filename = "popsize.jpg", width = 1200, height = 600,
##      res = 200)
par(mfrow = c(1, 2))
par(mar = c(2, 3.5, 0.5, 0.5))
boxplot(samples$meanFst ~ samples$size, axes = FALSE,
        xlab = "", ylab = "", main = "")
axis(side = 2, cex.axis = 0.7, tcl = -0.33, padj = 0.75)
mtext(side = 2, line = 2, expression("F"[D]))
axis(side = 1, cex.axis = 0.7, tcl = -0.33, padj = -0.75,
     labels = c("V Small", "Small", "Medium", "Large"),
     at = 1:4)
box()

par(mar = c(2, 3.5, 0.5, 0.5))
boxplot(samples$emlg ~ samples$size, axes = FALSE,
        xlab = "", ylab = "", main = "")
axis(side = 2, cex.axis = 0.7, tcl = -0.33, padj = 0.75)
mtext(side = 2, line = 2, "eMLG")
axis(side = 1, cex.axis = 0.7, tcl = -0.33, padj = -0.75,
     labels = c("V Small", "Small", "Medium", "Large"),
     at = 1:4)
box()

##dev.off()

```

# References
