---
title: "Population genetics of *Trichophorum planifolium* structured by inbreeding and geography, not environment"
output:
  bookdown::pdf_document2:
    toc: false
    number_sections: false
  bookdown::word_document2:
bibliography: plantarum.json
csl: american-journal-of-botany.csl
updated: 2021-02-08
---

# Abstract

## Premise

Peripheral populations are understood to hold special conservation
significance as repositories of divergent genetic diversity. They may also
be at increased risk of extirpation, due to a combination of small size,
isolation, and marginal habitat conditions. However, empirical studies
testing these expectations yield mixed results. We evaluated the
relationship between genetic diversity, geographical range limits, spatial
isolation, and environmental conditions. This work provides important
insight into the conservation of peripheral populations, and contributes to
our understanding of the factors shaping range limits.

## Methods

We used population samples, herbarium records, and climate data to survey
genetic diversity, geographic distribution, and environmental niche
characteristics of *Trichophorum planifolium* across its range in eastern
North America. Our analysis included distinct measures of geographic
peripherality, spatial isolation, and niche marginality, allowing us to
assess the influence of each of these factors on genetic diversity. 

## Results

Our data show that populations of *T. planifolium* have remarkably low
diversity, with the majority of populations having fewer than 3 distinct
multi-locus genotypes. While there was significant spatial structure, we
found no relationship between genetic diversity and peripherality,
isolation, or marginality.

## Conclusions

In *T. planifolium*, population genetic diversity is primarily a reflection
of local processes. Its reproductive biology favours inbreeding, and
genetic diversity is paritioned almost entirely among populations. This
highlights the importance of life history characteristics when
evaluating biogeographic patterns.

# Keywords
abundant center hypothesis; 
conservation genetics; 
microsatellites; 
*Trichophorum planifolium*; 
peripheral populations; 
range limits; 
species distribution models

# Introduction

No limit, aim for 8000 words.



```{R init, echo = FALSE, results = "hide"}
library(knitr)
knit("trich-prep.Rmd", quiet = TRUE)
```

In high-latitude countries, geographically peripheral populations form a
significant proportion of species-at-risk [@YakimowskiEckert_2007;
@GibsonEtAl_2009]. For example, in Canada, approximately 90% of threatened
or endangered plant species are locally rare representatives of species
that reach the northern extent of their range in Canada, but are common
further south [@YakimowskiEckert_2007]. Concern over the investment of
conservation resources into species that are locally rare but globally
common has led to debate over whether, or when, peripheral populations
merit national concern [e.g., @HunterHutchinson_1994; @HampePetit_2005;
@GibsonEtAl_2009].

Peripheral populations are typically small and geographically isolated due
to increased habitat variability and reduced resource availability at the
range margin [i.e., the abundant center hypothesis, @Brown_1984].
Consequently, they may be at greater risk of extirpation due to genetic
(i.e., inbreeding depression) as well as stochastic demographic,
environmental and catastrophic factors [@SouleMills_1998; @BrookEtAl_2002].
These risks may be exacerbated by anthropogenic threats such as habitat
loss and fragmentation [@YoungEtAl_1996]. Nevertheless, peripheral
populations may be important sources of adaptive and evolutionary potential
for the species [@SafrielEtAl_1994; @LeppigWhite_2006; @EckertEtAl_2008],
harbouring unique genetic diversity and/or adaptations to conditions at the
range edge [@LesicaAllendorf_1995; @BunnellEtAl_2004].

However, not all peripheral populations are genetically differentiated from
core populations [@LesicaAllendorf_1995; @LeppigWhite_2006]. In a review of
134 genetic studies in plants, [@EckertEtAl_2008] found that only 70.3% of
studies identified increased genetic differentiation towards the range
margin. These differences may be attributable to species-specific
characteristics [@GibsonEtAl_2009]. In plants, life-history characteristics
(i.e., breeding system and dispersal mechanisms) are a major determinant of
population genetic structure, because they are related to gene flow
[@LovelessHamrick_1984; @FrankelEtAl_1995]. In this regard, knowledge of
the level and spatial distribution of intraspecific genetic diversity can
add valuable information to guide conservation efforts for peripheral
populations, especially when the reproductive biology and ecology of the
species is not well understood.

@LesicaAllendorf_1995 proposed a framework for evaluating the conservation
value of peripheral populations, identifying population isolation and
divergent environmental conditions as main criteria. In practice, neutral
genetic markers, such as microsatellites, can be used to identify
populations that have experienced restricted gene flow (i.e., are
genetically isolated) [@CrandallEtAl_2000; @Moritz_2002]. However,
microsatellite markers cannot provide information on adaptive divergence of
populations [@HoldereggerEtAl_2006]. Therefore, to capture variation in
selection pressures across the species' range, environmental differences
among populations should be assessed [@AllendorfEtAl_2013]. 

These issues have received renewed attention recently, in light of new
approaches to assess the factors generally influencing range limits
[@CrossEckert_2020]. There is a growing recognition of the need to
distinguish between related spatial and ecological processes contributing
to the distribution limits of species [@WilliVanBuskirk_2019;
@Lee-YawEtAl_2018]. Three key concepts are frequently treated as loosely
interchangeable: geographic peripherality, geographic isolation, and
ecological marginality. However, the increasing availability of
high-resolution distribution and environmental data provides an opportunity
to examine the distinct influences of these processes on species range
limits.

*Trichophorum planifolium* (Sprengel) Palla, commonly known as Few-flowered
Club-rush or Bashful Bulrush, is a species that is rare at its northern
limit in Canada, but is common further south. It is a perennial woodland
species in the family Cyperaceae, typically occurring on mesic to dry,
often rocky, slopes in association with hardwoods, especially Oaks
(*Quercus* spp.) [@Crins_2002]. The range of *T. planifolium* extends from
Massachusetts west to Ontario and south to Virginia and Kentucky, with
disjunct populations occurring in Missouri and southern Illinois (Figure
\@ref(fig:distmap)).

In general, *T. planifolium* is considered common within the central
portion of its range, but rare along the periphery [@SmithRothfels_2007].
The only extant locality in Canada is in southern Ontario, in the nature
sanctuaries at Royal Botanical Gardens in Hamilton, Ontario. *Trichophorum
planifolium* was previously known from one other location in Ontario, in
Rouge Valley near Toronto, but no plants have been located at this site
since 2005 [@SmithRothfels_2007]. Due to its limited Canadian range and an
apparent decline in population size, *T. planifolium* is listed as
Endangered under the Canadian Species at Risk Act [SARA, @COSEWIC_2000]. 

The species is caespitose, growing in dense, low tufts (~ 40 cm) from short
rhizomes. It is wind-pollinated, possessing long exerted styles and stamens
and lacking both a showy perianth and nectaries [@Crins_1989;
@IwanyckiEtAl_2010]. Seeds mature in mid-summer and disperse in late July
to August [@IwanyckiEtAl_2010]. A mechanism for seed dispersal has not been
confirmed [@SmithRothfels_2007], but it has been suggested that the matting
of leaves and stems during the fruiting period may limit most seed
dispersal to within a meter of the parent plants, leading to the formation
of colonies of closely related sibling groups [@Crins_1989]. It has been
suggested that low recruitment observed at the Hamilton site may be due to
limited intrapopulational genetic diversity [@SmithRothfels_2007].

The main objective of our research was to assess genetic variability within
the Canadian population, and to quantify the relationship between the
geographic and ecological distribution of populations, and their relative
genetic diversity and divergence. Specifically, we address the questions:
does genetic diversity decline along spatial or environmental gradients
[@EckertEtAl_2008]; or does gene flow from the core of a species'
distribution maintain comparative levels of genetic diversity across the
range of the species [@KirkpatrickBarton_1997, @SextonEtAl_2016]? The
results of our work provide crucial insight for ongoing conservation
management of this endangered species.

# MATERIALS AND METHODS
## Population Sampling

In spring 2014, we sampled 29 populations of *T. planifolium* across its
geographic distribution (Figure \@ref(fig:distmap), Table
\@ref(tab:sumtab)). An additional population sample was previously obtained
in 2011 in Kentucky. At each site, we sampled individuals haphazardly, with
at least 1 m between individuals, and a maximum distance of 100 m between
the most distant samples. In total, 919 tissue samples were obtained, with
a minimum of four and a maximum of 50 samples per population (mean = 31.7).
Our protocol was modified for one very small population (PACR), where one
sample was taken at a distance of 150 m to increase sample size to seven. A
single herbarium voucher was collected per population, except for the
locations in Ontario, New Hampshire, and West Virginia, where the species
is Endangered and collection of whole plants is prohibited. Vouchers have
been deposited at DAO [abbreviations follow @Thiers_2020].

```{R distmap, fig.width = 7.25, fig.height = 3.5, dev = 'jpeg', fig.show = 'hold', dpi = 200, echo = FALSE, fig.cap = "*Trichophorum planifolium* distribution. A. Populations Sampled: Grey points are GBIF records used to generate the species distribution model and geographic isolation and marginality indices. Red diamonds are populations sampled for the genetic survey. B. Maxent species distribution model. Highest suitability is indicated in green, declining through yellow, orange and grey."}
par(mar = c(0.1,0.1,0.1,0), mfrow = c(1, 2))

plot(ca.proj, xlim = c(zm@xmin, zm@xmax),
     ylim = c(zm@ymin, zm@ymax), border = "grey",
     bg = "lightblue", col = "white")  
plot(us.proj, add = TRUE, border = "grey", col = "white")
plot(gl.proj, add = TRUE, border = "grey", col = "lightblue")
points(trich.proj, pch = 16, col = "grey30", cex = 0.7)
points(samples.proj, cex = 1, pch = 23, bg = 'red')
box()

text("A", x = grconvertX(0.1, from="npc", to="user"), 
     y = grconvertY(0.93, from = "npc", to="user"), cex = 2)

par(mar = c(0.1,0,0.1,0.1))

plot(ca.proj, xlim = c(zm@xmin, zm@xmax),
     ylim = c(zm@ymin, zm@ymax), border = "grey",
     bg = "lightblue", col = "white")  
plot(us.proj, add = TRUE, border = "grey", col = "white")
plot(predTR, add = TRUE, legend = FALSE)
plot(ca.proj, add = TRUE, border = "grey")
plot(us.proj, add = TRUE, border = "grey")
plot(gl.proj, add = TRUE, border = "grey", col = "lightblue")
box()

text("B", x = grconvertX(0.1, from="npc", to="user"), 
     y = grconvertY(0.93, from = "npc", to="user"), cex = 2)

```


```{R sumtab, results = 'asis', echo = FALSE}
sumTable <- as.data.frame(samples)
sumTable <- sumTable[, c("population", "Latitude", "Longitude", "n",
                        "NComp", "mlg", "emlg", "meanFst", "Ap", "Ar",
                        "Ho", "Hs", "Fis")]
sumTable["KY", "population"] <- "KYLE"
rownames(sumTable) <- NULL

tmp <- sumTable[1,]
tmp[, names(total)] <- total
tmp[, "population"]  <- "Total"
sumTable <- rbind(sumTable, tmp)
sumTable[sumTable$population == "Total",
         !colnames(sumTable) %in% names(total)] <- NA

options(knitr.kable.NA = '')

sumTable %>%
  mutate(Latitude = round(Latitude, 4),
         Longitude = round(Longitude, 4),
         N = paste(n, "(", NComp, ")", sep = ""),
         emlg = round(emlg, 2),
         Fst = round(meanFst, 2),
         Ar = round(Ar, 2),
         Ho = round(Ho, 2),
         Hs = round(Hs, 2),
         Fis = round(Fis, 2)
         ) %>%
  select(population, Latitude, Longitude,
         N, Ap, Ar, mlg, emlg, Hs, Ho, Fis, Fst) %>%
  kable(caption =
          "Populations sampled. The first two letters of the Population Code indicate the US state or Canadian province. N: total samples per population (samples with all 11 loci scored). Ap: private alleles. Ar: mean allelic richness per locus. mlg: multilocus genotypes. eMLG: multilocus genotypes in rarefied samples of 10 individuals. Hs: gene diversity. Ho: observed heterozygosity.") 
```

## Genetic Analysis

After collection, samples were immediately placed on silica gel and stored
at room temperature prior to molecular analysis. A random subsample of 20
samples per population were selected for genetic analysis. For populations
with fewer than 20 samples, all individuals were used.

For each individual, 20 mg of dried leaf tissue was ground with 3
mm-diameter stainless steel beads in a 2 mL Eppendorf tube at 30 Hz for 2
min using a TissueLyser II (Qiagen, Venlo, Limburg, Netherlands). Total
genomic DNA was extracted from ground tissue using the Nucleospin Plant II
Kit (Machery-Nagel, Düren, Germany). The quality and concentration of DNA
obtained was verified using a Nanodrop 2000 UV-Vis Spectrophotometer
(ThermoScientific, Waltham, Massachusetts, U.S.A.).

We developed eleven nuclear microsatellite markers [see @NowellEtAl_2015
for details] to assess the population genetics of *T. planifolium*.
Amplification of microsatellite fragments were performed in 8 µl reaction
volumes containing 0.416 µL primer mix (10 mM; 10:1 untagged to tagged
primer), 0.192 µL dye-labelled CAG Tag (10 mM; 6-FAM or VIC, Life
Technologies, Carlsbad, California, U.S.A.), 0.24 µL DMSO, 4 µL 2X Phusion
High-Fidelity Master Mix with HF Buffer (New England Biolabs, Ipswich,
Massachusetts, U.S.A.), 2.152 µL ddH2O and 1 µL DNA (10 ng/µL) using a
T-100 Thermal Cycler (Bio-Rad, Hercules, California, U.S.A.). Cycling
conditions followed Touchdown-TD PCR [@KorbieMattick_2008] as in
@NowellEtAl_2015. PCR products were visualized on a 1% agarose gel
stained with GelRed (Biotium, Hayward, California, U.S.A.) and viewed with
a High Performance Transilluminator (UVP, Upland, California, U.S.A.) with
a 100 bp DNA ladder (New England Biolabs) to confirm the presence and size
of amplification products and absence of contamination prior to genotyping.

Amplification products were subsequently pooled into groups of four and
visualized by capillary electrophoresis using a 3130xL Genetic Analyzer
(Life Technologies) with the GeneScan 500 LIZ Size Standard (Life
Technologies). Individual samples were genotyped using Genemapper v.5
software (Life Technologies) and verified with manual scoring.

Due to amplification failures, we were unable to score all samples for all
11 markers. We retained only those samples for which we could score at
least 8 loci. We enumerated private alleles, calculated allelic richness
[@ElMousadikPetit_1996; @AdamackGruber_2014], gene diversity, observed
heterozygosity and the fixation index F~IS~ [@Nei_1987;
@GoudetJombart_2015] for each population. Tests for departures from
Hardy-Weinberg equilibrium (HWE) were performed with the R package 'pegas'
ver. 0.13 [@Paradis_2010].

We counted the number of distinct multi-locus genotypes (MLG) for each
population, using the methods provided by @KamvarEtAl_2015. When samples
with incomplete genotypes could not be unambiguously assigned to a single
MLG, we randomly assigned them to a matching MLG in same population, if
present. To account for differences in sample size, we also calculated
rarefied MLG counts for each population, as a standardized measure of
genetic richness. The rarefaction estimates were calculated for subsamples
of 10 individuals per population [referred to as eMLG hereafter,
@OksanenEtAl_2019]. Populations with fewer than 10 invidivduals scored were
excluded from rarefaction, and all analyses based on eMLG.

We also assessed genetic divergence, using an index based on population
pairwise-F~ST~ values. We calculated our divergence index (F~D~) as the
mean of pairwise-F~ST~ values for a given population, when compared with
all other populations.

Preliminary review of the data revealed excessive levels of homozygosity,
and no locus was in Hardy-Weinburg equilibrium. As discussed below, this
indicates high levels of selfing and/or inbreeding among close relatives.
As such, analyses based on assumptions of Hardy-Weinburg equilibrium,
linkage equilibrium and panmictic populations, such as STRUCTURE
[@PritchardEtAl_2000], are inappropriate for our data. Accordingly, we used
multivariate ordinations to visualize genetic structure among populations
[@JombartEtAl_2008]. Similarly, we did not test for the presence of null
alleles, as we are not aware of a test that can distinguish between
excessive homozygosity due to the presence of null alleles, and excessive
homozygosity to due to high levels of inbreeding [e.g., @Brookfield_1996].

## Geographic and Ecological Analysis

We assessed spatial structure using spatial Principal Components Analysis
[i.e., sPCA; @JombartEtAl_2008; @ThioulouseEtAl_2018]. We defined the
spatial neighbourhood of a sample as all other samples within 650 km, with
the strength of each link weighted inversely by distance (via function
`spdep::dnearneigh` with the upper distance bound set to 650). We chose 650
km to ensure there were no disconnected subgroups (i.e., there was at least
one link between the Missouri populations and the eastern populations). We
tested for statistically significant spatial autocorrelation using a
randomization test (function `global.rtest` in R package `adegenet`).

Additionally, we used eMLG and F~D~ to assess the relationship between
genetic diversity and three indexes representing geographic and ecological
distribution: Isolation (the distance between a population and its nearest
neighbours), Peripherality (the relative spatial location of a population
relative to the core distribution of the species), and Suitability (the
relative ecological suitability of a population's location). 

We generated all three of these indices using specimen records downloaded
from @GBIF.org_2019 to characterize the geographic distribution of *T.
planifolium*. To control for collection bias [see
@RadosavljevicAnderson_2014], we used the `spThin` package
[@Aiello-LammensEtAl_2015] to create a thinned data set, in which retained
samples were at least 10 km from all other samples.

We calculated the isolation index as the mean geographic distance between a
sampled population and its five nearest neighbours in the thinned GBIF data
set. We decided to use five nearest neighbours, rather than the single
nearest neighbour, as this would score isolated clusters of 2-4 populations
as having greater isolation than a similar sized clusters that occur in
close proximity to other populations.

We calculated the peripherality index as the Mahalanobis distance between
each sample and the centroid of the GBIF data. We used the Mahalanobis
distance, rather than the Euclidean distance, as the former accounts for
the density & distribution (variance and covariance) of points, as well as
their absolute distances [@LegendreLegendre_1998; see @Lee-YawEtAl_2018 for
a similar approach].

The suitability index was generated using a Maxent distribution model
[@PhillipsEtAl_2017]. We limited the study extent to 31N-50N
degrees latitude, 66W-98W degrees longitude; this area was chosen to
include a 500 km buffer around a minimum convex polygon containing all GBIF
records for *T. planifolium*. Background training data was limited to level
1 Ecoregions [@OmernikGriffith_2014] in which *T. planifolium* is present
within this extent. The following environmental layers were considered: 19
bioclimatic variables [WorldClim: @FickHijmans_2017, www.worldclim.org],
aridity [@TrabuccoZomer_2019], and terrain ruggedness. Terrain ruggedness
was calculated from SRTM DEM data [@JarvisEtAl_2008] with 3-arc second
(approximately 90m) horizontal resolution. We used the Terrain Ruggedness
Index in QGIS [@QGIS.org_2020]. The resulting raster was upscaled to match
the 30 arc-second (~1km) resolution of the other layers, using the
"average" resampling method. To reduce collinearity among variables, We
selected a subset of these layers such that the variance inflation factor
was less than 10 [function `vif` in the R package `usdm`: @NaimiEtAl_2014].

We tested the relationships between genetic diversity (eMLG) and genetic
divergence (F~D~), and Isolation, Peripherality, and Suitability using
linear regression. The index values were log-transformed to improve
normality.

All analyses were performed in R version 4.0 [@RCoreTeam_2020], unless
otherwise specified.

# RESULTS
We scored a total of 505 individual genotypes for at least 8 loci, and
completed all 11 loci for 408 individuals (Table \@ref(tab:sumtab)). 46
alleles were detected (mean 4.2 per locus, range 2--9), and a total of 115
unique multi-locus genotypes (MLG) were described. All but three MLG were
confined to a single population. One MLG was shared by two Missouri
populations; one was shared by two Ohio populations; and one MLG was found
in three populations, one each in Connecticut, Maryland, and eastern New
York.

465 (92.1%) individuals were homozygous for all alleles, with only 40
individuals heterozygous at one or more loci. Of the 25 populations with at
least 10 individuals genotyped, the mean eMLG (i.e., genotypes per 10
individuals) was 3.1 (range 1--6.5, Figure \@ref(fig:sumplots)). The
Ontario population had an eMLG of 1.37, corresponding to the 20th
percentile of the distribution. The mean F~D~ was 0.79 (range 0.59--0.9,
Figure \@ref(fig:sumplots)); F~D~ for the Ontario population was 0.87,
corresponding to the 80th percentile of the distribution.

```{R sumplots, fig.width = 7, fig.height = 3.5, dev = 'jpeg', dpi = 200, echo = FALSE, fig.show = "hold", fig.cap = "Distribution of rarified multi-locus genotypes (eMLG) and population mean pair-wise Fst values. The red line indicates the Ontario population.", }
#dev.new(height = 3.5, width = 7)

par(mfrow = c(1, 2))
par(mar = c(3.5, 3.5, 0.5, 0.5))
hist(samples$emlg, breaks = 12, main = "", xlab = "", ylab = "",
     axes = FALSE) 
abline(v = data.frame(samples)["ONRB", "emlg"], col = 2, lwd = 4)

axis(side = 2, cex.axis = 0.7, tcl = -0.33, padj = 0.75)
mtext(side = 2, line = 2, "Frequency")
axis(side = 1, cex.axis = 0.7, tcl = -0.33, padj = -0.75)
mtext(side = 1, line = 2, "eMLG")

par(mar = c(3.5, 1.5, 0.5, 0.5))
hist(samples$meanFst, breaks = 12, main = "", xlab = "",
     ylab = "", axes = FALSE)
abline(v = data.frame(samples)["ONRB", "meanFst"], col = 2, lwd = 4)

axis(side = 2, cex.axis = 0.7, tcl = -0.33, padj = 0.75)
axis(side = 1, cex.axis = 0.7, tcl = -0.33, padj = -0.75)
mtext(side = 1, line = 2, text = expression('F'[D]))

```

There was significant spatial structure in the genetic data, as determined
by a Monte Carlo test of Morin's I (function `adespatial::global.rtest`, p
< 0.001). Plotting the first axis of the sPCA on a geographic map reveals
this structure is largely due to divergence of the Missouri populations
from the eastern locations (Figure \@ref(fig:sPCAmap)). However, the
Ontario population and one of the Ohio populations are genetically more
similar to the Missouri populations than to their nearer neighbours.

```{R sPCAmap, fig.width = 7, fig.height = 5, dev = 'jpeg', dpi = 200, echo = FALSE, fig.show = "hold", fig.cap = "*Trichophorum planifolium* Populations Sampled. Colour and size of points indicate the position of each population on sPCA axis 1: size indicates absolute distance from the origin, and colour indicates positive or negative values. i.e., large black squares have large negative values, large white values have large positive PCA values, and smaller squares are have lower values."}
##dev.new(height = 5, width = 7)
par(mar = c(0.1,0.1,0.1,0.1))

###axcol <- 2 * (trich.spca1$li[, 1] - min(trich.spca1$li[, 1])) + 1
###axcol <- axcol/min(axcol)
###axcol  <- axcol < 0.5

med <- mean(range(trich.spca1$li[,1]))
axex <- trich.spca1$li[,1] - med
axex <- abs(axex)
axex <- 4 * (axex + 0.25)
axcol <- trich.spca1$li[, 1] < med

plot(ca.proj, xlim = c(zm@xmin, zm@xmax),
     ylim = c(zm@ymin, zm@ymax), border = "grey",
     bg = "lightblue", col = "grey95")  
plot(us.proj, add = TRUE, border = "grey", col = "gray95")
plot(gl.proj, add = TRUE, border = "grey", col = "lightblue")
##points(samples.proj, cex = axex, pch = 22, bg = grey(axcol/max(axcol)),
##       col = "grey")
points(samples.proj, cex = axex, pch = 22,
       bg = c("black", "white")[axcol+1], col = "darkgrey")

box()
```

We found no evidence supporting a relationship between genetic diversity or
genetic divergence and geographic location (isolation or peripherality) or
ecological condition (suitability; Table \@ref(tab:regression)).


```{R regression, echo = FALSE}
lmp <- function (modelobject) {
  if (class(modelobject) != "lm") stop("Not an object of class 'lm' ")
  f <- summary(modelobject)$fstatistic
  p <- pf(f[1],f[2],f[3],lower.tail=F)
  attributes(p) <- NULL
  return(p)
}

lms <- data.frame(response = rep(c("emlg", "meanFst"), each = 3),
                 explanatory = rep(c("isolation", "marginality",
                                     "suitability"), 2),
                 F = NA, R = NA, p = NA)

for(i in seq_along(lms[, 1])){
  form <- formula(paste(lms$response[i],
                       "~ log(", lms$explanatory[i], ")"))
  mod <- lm(formula = form,
           data = samples@data)
  lms[i, "R"] <- round(summary(mod)$adj.r.squared, 2)
  lms[i, "p"] <- round(lmp(mod), 2)
  lms[i, "F"] <- round(summary(mod)$fstatistic[1], 2)
}

# make sure our distance matrices are aligned:
stopifnot(row.names(as.matrix(trichFst)) == row.names(as.matrix(geogD)))

ibd <- mantel(trichFst, as.dist(geogD))
lms <- lms[c(1, 1:6), ]
lms[1, 1]  <- "IBD"
lms[1, 2]  <- ""
lms[1, "R"] <- round(ibd$statistic, 2)
lms[1, "p"] <- round(ibd$signif, 2)
lms[1, "F"] <- NA

colnames(lms)[1:2] <- c("Dependent", "Independent")
lms[c(3, 4, 6, 7), 1] <- ""
lms[2, 1] <- "eMLG ~"
lms[5, 1] <- "Fst ~"
rownames(lms) <- NULL
## ibdMat <- data.frame(Fst = as.numeric(trichFst),
##                     distance = as.numeric(as.dist(geogD)))
## ggplot(ibdMat) + aes(x = log(Fst), y = distance) +
##   geom_point() + geom_smooth(method = "lm")

lms <- lms[-1, ]
kable(lms,
      caption =
      "Biogeographic Tests. Each line reports linear regression parameters F-statistic, adjusted R-squared value and P-value. Dependent variables: eMLG, rarefied richness of multi-locus genotypes; Fst, mean pairwise Fst values for each population. Independent variables: indexes of isolation, peripherality (see text), and habitat suitability from SDM.")
```

# DISCUSSION
We found no evidence to support the hypothesis that genetic diversity
declines with geographic peripherality, ecological marginality, or spatial
isolation. However, the highly structured populations (F_ST = 0.8) and
significant east-west spatial structure suggest that this lack of pattern
is not due to 'swamping' gene flow from the core of the distribution
homogenizing genetic diversity in peripheral populations.

This contrasts with other recent studies that have found significant
relationships between genetic diversity and spatial or environmental
gradients [@SextonEtAl_2016; @Lee-YawEtAl_2018]. Differences in life
history and reproductive biology may explain the difference.
@Lee-YawEtAl_2018 found that genetic diversity declined towards the
geographic range limits of *Arabidopsis lyrata* ssp. *lyrata*, and also
with distance from the species' environmental niche. In addition, their
data shows that environmental conditions declined towards the range
periphery. This supports the abundant center hypothesis [@Brown_1984]. 

On the other hand, @SextonEtAl_2016 found no relationship between genetic
diversity and spatial structure in *Mimulus laciniatus*. Instead, their
work documented Isolation by Environment [@SextonEtAl_2014], where genetic
diversity is structured along environmental gradients, presumably a
consequence of natural selection or non-random mating.

Neither pattern appears in our data. The most striking observation in our
study is the extremely low variation present within populations (mean FIS
0.91, 1% heterozygosity), and high level of differentiation among
populations (Fst = 0.80). *Trichophorum planifolium* is apparently a
wind-pollinated, outcrossing species, but these values are higher even than
typical for selfing species [i.e., the mean Fst for selfing species with
similar life history traits is 0.4--0.57, following @HamrickGodt_1996;
@Nybom_2004].

This supports Crins' [-@Crins_1989] hypothesis that the low stature of *T.
planifolium*, and the complex terrain it occupies, limits outcrossing.
Crins also noted that *T. planifolium* is protandrous, presumably an
adaptation to lessen the frequency of self-pollination. However,
@FriedmanBarrett_2009 showed that the spatial and temporal separation of
male and female flowers in woodland sedges (monecious *Carex*) are largely
ineffective at preventing geitonogamy. They studied seven *Carex* species,
many of which co-occur with *T. planifolium* in woodland habitats and have
similar caespitose habit. They found only one species had an outcrossing
rate greater than 0.1 (three species had no polymorphic allozymes, which
may reflect a long history of inbreeding).

Lacking any obvious adaptations for long-distance seed dispersal, gene flow
via seed dispersal is likely similarly limited. It is well documented that
plant seeds often travel only one or a few metres [@CainEtAl_2000]. Seed
dispersal in *T. planifolium* occurs in late July and August, during which
time the plants' leaves become matted. Crins [-@Crins_1989] suggested that
this would facilitate deposition of the seeds immediately adjacent to
parent plants (i.e., seeds are gravity dispersed), resulting in small
spatial clumps of closely related tussocks. Like self-pollination, mating
among closely related individuals limits recombination and leads to
increased homogeneity of genotypes in the population [@LovelessHamrick_1984].

In combination, these life history characteristics may explain the low
population genetic diversity documented here as the outcome of founder
effects, followed by population expansion dominated by selfing. This is
consistent with a general pattern noted in other sedges (*Carex*), which
can be divided into two life history categories [@FordEtAl_1998]. "Group 1"
are caespitose, with multiple hermaphroditic culms on each plant, with
flowers in close proximity. This facilitates geitonogamy, resulting in low
intra-population diversity, but high inter-population diversity [e.g.,
@Waterway_1990; @SchellWaterway_1992; @Whitkus_1992, @KullOja_2007].

In contrast, "Group 2" *Carex* are rhizomatous, with more space between
culms from the same individual, and more physical mixing among individuals.
This leads to greater intra-population diversity, and less differentiation
among populations [@FordEtAl_1998].

*Trichphorum planifolium* fits the description of *Carex* Group 1, aside
from having bisexual flowers. This likely further exacerbates inbreeding,
as it may engage in both autogamous and geitenogamous selfing. The low
levels of heterozygosity we observed (Ho 0.00--0.03, Table
\@ref(tab:sumtab)) are comparable to values reported for other sedges:
*Trichophorum caespitosum*: Ho = 0.01 to 0.04 [@GodtEtAl_1996], *Carex
loliaceae*: Ho = 0.00 to 0.12 [@KullOja_2007] and *Carex magellanica*
subsp. *irrigua*: Ho = 0.00 to 0.03 [@KullOja_2010]. 

The main geographic structure in the genetic data reveals a broad division
between the Missouri and the eastern populations (Figure
\@ref(fig:distmap)). This suggests current *T. planifolium* populations may
have emerged from glacial refugia both in the Appalachian Mountains and on
the Ozark plateau [@DelcourtDelcourt_1991; @Meyer_1997; @SoltisEtAl_2006].
However, if this is the case, it is interesting to note that populations
representing both eastern and western genetic groups appear in close
proximity in Ohio. The sPCA analysis also indicates the Ontario population
is genetically more similar to the Missouri populations than the
geographically closer eastern group.

Gene flow via pollen or seed between Missouri in the west, and Ohio and
Ontario to the east, is unlikely, given the long distances between these
locations (> 500 km). However, the SDM indicates pockets of suitable
environment exist along the Ohio river in southern Indiana and northern
Kentucky (Figure \@ref(fig:distmap))). This may have served as a
post-glacial colonization route linking populations from the two refugia;
there may even be overlooked populations of *T. planifolium* in this area,
as this species is easily overlooked.

While our data do not show a general relationship between spatial or
environmental conditions and genetic diversity, the lone remaining Canadian
population is among the least diverse (eMLG; Figure \@ref(fig:sumplots)).
We have argued above that population dynamics in this species are
characterized by bottlenecks followed by inbreeding. This is often
associated with a reduction in fitness [@AngeloniEtAl_2011]. However, this
cost may be reduced in self-pollinating species [@Busch_2005;
@HoltsfordEllstrand_1990], and can be further mitigated if the inbreeding
leads to purging of deleterious alleles [@CrnokrakBarrett_2002].

We cannot address this issue directly with our data. Microsatellites are
neutral markers, they do not provide direct evidence of adaptive diversity;
they only provide insight into gene flow and reproductive strategies. We
can say that *T. planifolium* populations with low diversity are relatively
common and widespread (e.g., CT, MA, OH, VA), and several of
them are quite large and apparently stable (personal observation).
Furthermore, this pattern of limited intra-population genetic diversity is
common in other woodland sedges. From this, we conclude that inbreeding
depression is not likely to represent a conservation concern in this case.

The Canadian population is also one of the most divergent (F~D~; Figure
\@ref(fig:sumplots)). This provides indirect evidence that this population
is adapted to site-specific environmental conditions, and indicates that it
is of high conservation value [as per @LesicaAllendorf_1995]. Indeed,
protection of many populations of *T. planifolium* would be necessary to
conserve species-level genetic diversity as it is chiefly partitioned
among, rather than within, populations [see @BaskaufEtAl_2014 or
@KoellingEtAl_2011, for example]. These results also suggest that risks of
outbreeding depression are high with outcrossing, and future restorative
efforts (i.e., augmentation) for the Canadian population should be avoided
or approached with caution [@FrankhamEtAl_2011]. It should also be
considered that mixing of populations with differing genetic backgrounds
could decrease the conservation value of the Canadian population
[@BarrettKohn_1991].

# Acknowledgements
The authors thank The staff of Royal Botanical Gardens in Hamilton,
Ontario, especially Natalie Iwanycki, who provided detailed reports on *T.
planifolium* in Canada. Kasia Dadej (Agriculture and Agri-Food Canada)
assisted with scoring the microsatellite data. The following people
assisted with obtaining access to field sites: George Yaskievych, Rick
Gardner, Tom Wieboldt, P. J. Harmon, Jim Vanderhorst, John Kunsman, Jeff
Saarela, Steve Grund, Steve Glenn, Sara Cairns, Bryan Connolly, Robert
Bertin, Karen Searcy, Rich Ring, Steven Daniel, Greg Iffrig, Brian
Davidson, April Daniels, Dale Egbert, Lynn Prater, Nathan Jester, Rich
McCarty, Dave Minney, Bill Stanely, Phil Hoffman, Glen Cobb, Levi Miller,
Pearl's family, Bruce Metz, Fawn Kearns, Barbara Sargent, Tom Metzger, the
Morris County Parks Commission, Becky Petro, Julie Richburg, Nancy Putnam,
Jesse Jaycox, and Mark Quinn. Funding for this project was provided by the
Ontario Ministry of Natural Resources Species at Risk Stewardship Fund
(SAR-SF), the Environment and Climate Change Canada Interdepartmental
Recovery Fund (IRF), and the Carleton University Bonneville Family
Scholarship.

# AUTHOR CONTRIBUTIONS

VJN designed the majority of this study as part of her MSc thesis research.
She planned the field sampling, conducted all the field work, completed the
lab work, and wrote the first draft of the manuscript.

TWS supervised VJN throughout her MSc program, contributing to the study
design, partipated equally in field work, assisting with data analysis and
interpretation, and revising the manuscript. He also contributed additional
spatial analyses, and prepared the final draft of the manuscript.

SW assisted VJN in designing the molecular markers used in this project,
and participated equally in completing the lab work.

# References
